src/overlays/actors/ovl_Door_Ana/z_door_ana.c

```diff
@@ -22,19 +33,184 @@ const ActorInit Door_Ana_InitVars =
+            ActorCollider_AllocCylinder(globalCtx, collider);
+            ActorCollider_InitCylinder(globalCtx, collider, &this->actor, &colliderInit);
+        }
+        else
+        {
+            this->actor.flags |= 0x10;
+        }
+        Actor_SetScale(&this->actor, 0);
+        func_80993EF0(this, (ActorFunc)&func_80993FEC);
+        this->actor.unk_1F = 0;
+        return;
+    }
+    func_80993EF0(this, (ActorFunc)&func_80994124);
+    this->actor.unk_1F = 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Ana/func_80993FEC.s")
+static void DoorAna_Destroy(DoorAna* this, GlobalContext* globalCtx)
+{
+    // free collider if it has one
+    if ((this->actor.params & 0x200) != 0)
+    {
+        ActorCollider_FreeCylinder(globalCtx, &this->collider);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Ana/func_80994124.s")
+// update routine for grottos that are currently "hidden"/unopened
+static void func_80993FEC(DoorAna* this, GlobalContext* globalCtx)
+{
+    bool openGrotto = false;
+    if ((this->actor.params & 0x200) == 0)
+    {
+        // opening with song of storms
+        if (this->actor.waterSurfaceDist < 40000.0f && func_8006C4A4(globalCtx, 5) != 0)
+        {
+            openGrotto = true;
+            this->actor.flags &= -0x11;
+        }
+    }
+    else
+    {
+        // bombing/hammering open a grotto
+        if ((this->collider.base.collideFlags & 2) != 0)
+        {
+            openGrotto = true;
+            ActorCollider_FreeCylinder(globalCtx, &this->collider);
+        }
+        else
+        {
+            ActorCollider_Cylinder_Update(&this->actor, &this->collider);
+            Actor_CollisionCheck_SetAC(globalCtx, &globalCtx->sub_11E60, &this->collider);
+        }
+    }
+    // open the grotto
+    if (openGrotto)
+    {
+        this->actor.params &= 0xFCFF;
+        func_80993EF0(this, (ActorFunc)&func_80994124);
+        Audio_PlaySoundGeneral(0x4802, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+    }
+    func_8002F5F0(&this->actor, globalCtx);
+}
+
+// update routine for grottos that are open
+static void func_80994124(DoorAna* this, GlobalContext* globalCtx)
+{
+    Player* player;
+    s32 destinationIdx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Ana/func_809942D8.s")
+    player = PLAYER;
+    if (Math_ApproxF(&this->actor.scale.x, 0.01f, 0.001f) != 0)
+    {
+        if ((this->actor.unk_1F != 0) && (globalCtx->sceneLoadFlag == 0) && 
+            (player->stateFlags1 & 0x80000000) && (player->unk_84F == 0))
+        {
+            destinationIdx = ((this->actor.params >> 0xC) & 7) - 1;
+            func_800C0AF4(globalCtx, 1, 0x4FF);
+            gSaveContext.respawn[RESPAWN_MODE_RETURN].pos.y = this->actor.posRot.pos.y;
+            gSaveContext.respawn[RESPAWN_MODE_RETURN].yaw = this->actor.initPosRot.rot.y;
+            gSaveContext.respawn[RESPAWN_MODE_RETURN].data = this->actor.params & 0xFFFF;
```

Comment:
Feel like the & FFFF is probably not needed or part of some kind of cast. If its needed though then no big deal